SimpleWAWS/Code/CsmExtensions/CsmManager.cs

```diff
@@ -36,8 +37,20 @@ static CsmManager()
             graphClient = new AzureClient(retryCount: 3);
             graphClient.ConfigureUpnLogin(SimpleSettings.TryUserName, SimpleSettings.TryPassword);
 
-            acomClient = new AzureClient(retryCount: 3);
-            acomClient.ConfigureSpnLogin(SimpleSettings.TryDevOpsTenant, SimpleSettings.TryDevOpsServicePrincipal , SimpleSettings.TryDevOpsServicePrincipalKey);
+            jenkinsClient = new AzureClient(retryCount: 3);
+            jenkinsClient.ConfigureSpnLogin(SimpleSettings.JenkinsTenant, SimpleSettings.JenkinsServicePrincipal , SimpleSettings.JenkinsServicePrincipalKey);
+        }
+
+        static AzureClient GetClient(Subscription.SubscriptionType subscriptionType)
```

Comment:
It sounds interesting. So the call will be `wrappedCsmClient.HttpInvoke(HttpMethod.Get, ArmUriTemplates.CsmTemplateDeployment.Bind(csmDeployment))`  and the wrapper will parse the `subscriptionId` from the `Uri` determine the subType and use an existing static client of those type. We could make wrappedCsmClient a DynamicObject with a sync and async `HttpInvoke` Methods that do the needful. 
Any other ideas ?
I'll test this out